src/attach.js

```diff
@@ -16,8 +16,19 @@ function interceptor(fetch, ...args) {
     }
   });
 
+  let request = null;
+
   // Register fetch call
-  promise = promise.then(args => fetch(...args));
+  promise = promise.then(args => {
```

Comment:
Is this safe in race conditions? i.e. what happens if a second `fetch` is started before the first one returns

Reply:
Thats correct we need to nest the a second promise that extends the `response` object. Then you can use `const request` instead of `let` which gives additional safety in this regard.